### PR TITLE
MI-2269: Switch

### PR DESCRIPTION
**Pull request by tanisha-agarwal**
### Ticket Link
 [MI-2269](https://brightscout.atlassian.net/browse/MI-2269)

### Description
- Created Switch Component

### Screenshots
image
image
image
image
image
image
image

### CODE REVIEW

File: src/components/Switch/Switch.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import {Switch} from './Switch.component';
+import {SwitchProps} from './Switch';
+
+export default {
+    title: 'Component/Switch',
+    component: Switch,
+} as Meta;
+
+const SwitchTemplate:Story<SwitchProps> = (args) => <Switch {...args}/>;
```
Comment (raghavaggarwal2308) [inline src/components/Switch/Switch.stories.tsx:12]:
```suggestion
const SwitchTemplate: Story<SwitchProps> = (args) => <Switch {...args}/>;
```

**Review by raghavaggarwal2308 — APPROVED** [review]

File: src/components/Switch/Switch.d.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * Interface for Styled Switch Container
+ */
+export interface StyledSwitchContainerProps {
+
+    /**
+		* if 'true', Toggle switch occupies full width of it's container
+		*
+		* @default false
+		*/
+    fullWidth?:boolean;
```
Comment (ayusht2810) [inline src/components/Switch/Switch.d.ts:11]:
```suggestion
    fullWidth?: boolean;
```

Comment (ayusht2810) [reply src/components/Switch/Switch.d.ts:11]: check everywhere

File: src/components/Switch/Switch.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+// Error Switch
+export const Error = SwitchTemplate.bind({});
+Error.args = {
+    label: 'Label',
+    error: true,
+};
+
+// Error Switch
```
Comment (ayusht2810) [inline src/components/Switch/Switch.stories.tsx:55]:
```suggestion
// Full Width Switch
```

**Review by ayusht2810 — APPROVED** [review]

File: src/components/Switch/Switch.component.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            />
+            <div className='mm-switch_labels'>
+                <Form.Check.Label className='mm-switch_label'>{label}</Form.Check.Label>
+                <Form.Check.Label className='mm-switch_subLabel'>
+                    {subLabel}
+                </Form.Check.Label>
+            </div>
+        </StyledSwitch>
+    );
+};
```
Comment (avas27JTG) [inline src/components/Switch/Switch.component.tsx:45]:
```suggestion
export const Switch = ({id, label, subLabel, error, className = '', fullWidth = false, ...restProps}: SwitchProps) => (
        <StyledSwitch
            className={`mm-switch ${className}`}
            id={id}
            fullWidth={fullWidth}
        >
            <Form.Check.Input
                className='mm-switch_input'
                isInvalid={error}
                {...restProps}
            />
            <div className='mm-switch_labels'>
                <Form.Check.Label className='mm-switch_label'>{label}</Form.Check.Label>
                <Form.Check.Label className='mm-switch_subLabel'>
                    {subLabel}
                </Form.Check.Label>
            </div>
        </StyledSwitch>
    );
```

see if this works

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**